From 9e22afa48d5fe782df6d2af00cdd42930cd7ac03 Mon Sep 17 00:00:00 2001
From: Shreyat Pandey <quic_shrepand@quicinc.com>
Date: Mon, 16 Sep 2024 15:13:05 +0530
Subject: [PATCH] Fix STA's SSID protection capability when AP SME is offloaded
 to driver

Correctly set STA's SSID protection capability to STA's wpa_sm for AP
SME offload to the driver case when processing association events.

Signed-off-by: Shreyat Pandey <quic_shrepand@quicinc.com>
---
 src/ap/drv_callbacks.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/ap/drv_callbacks.c b/src/ap/drv_callbacks.c
index ab76a5fc6..ba54fd792 100644
--- a/src/ap/drv_callbacks.c
+++ b/src/ap/drv_callbacks.c
@@ -678,6 +678,13 @@ int hostapd_notif_assoc(struct hostapd_data *hapd, const u8 *addr,
 			goto fail;
 		}
 #endif /* CONFIG_SAE */
+
+		wpa_auth_set_ssid_protection(
+			sta->wpa_sm,
+			hapd->conf->ssid_protection &&
+			ieee802_11_rsnx_capab_len(
+				elems.rsnxe, elems.rsnxe_len,
+				WLAN_RSNX_CAPAB_SSID_PROTECTION));
 	} else if (hapd->conf->wps_state) {
 #ifdef CONFIG_WPS
 		struct wpabuf *wps;
-- 
2.43.0

